### PR TITLE
aosp.py: Update support of several important branches

### PR DESCRIPTION
**Pull request by legendlee1314**
Supported lists:
No.    Branch    Product(x86, x86_64)
1.     irdal     cohol
2.     gminl     ecs_e7(,_64p)
3.     stable    ecs_e7(,_64p)
4.     upstream  mako

workaround for aosp-stable:
* fastboot not ready so use a prebuilt loader.efi and fastboot.img.
* use specific flash.sh.
* all can be downloaded at wp-02:/workspace/server/workaround.

### CODE REVIEW

**Comment by legendlee1314** [comment]
@gyagp , please review this patch.

File: python/aosp.py [hunk 1]
```diff
@@ -287,7 +287,9 @@ def build():
 
         combo = _get_combo(target_arch, target_type)
         if repo_type == 'upstream':
-            dir_driver_upstream = '/workspace/topic/aosp/driver'
+            dir_driver_upstream = dir_share_python + '/aosp/driver'
+            if not os.path.exists(dir_driver_upstream):
+                error('Directory %s do not exist' % dir_driver_upstream)
```
Comment (gyagp) [inline python/aosp.py]: Just omit this check as it will check dir_driver_upstream_one later on.

File: python/aosp.py [hunk 2]
```diff
@@ -421,26 +423,34 @@ def flash_image():
...
             elif repo_type == 'irdakk':
                 file_image = dir_root + '/out/target/product/irda/irda-ktu84p-factory.tgz'
+            elif repo_type == 'irdal':
+                file_image = dir_root + '/out/dist/%s-lrx22c-factory.tgz' % product
             elif repo_type == 'gminl':
-                file_image = dir_root + '/out/target/product/%s_%s/%s_%s-lrx21n-factory.tgz' % (product_brand, product_name, product_brand, product_name)
+                file_image = dir_root + '/out/dist/%s-lmy07b-factory.tgz' % product
             elif repo_type == 'gminl64':
-                file_image = dir_root + '/out/target/product/%s_%s_64p/%s_%s_64p-lrx21n-factory.tgz' % (product_brand, product_name, product_brand, product_name)
+                file_image = dir_root + '/out/dist/%s-lmx86-factory.tgz' % product
```
Comment (gyagp) [inline python/aosp.py]: We need to make approximate matching with 'lrx22c', 'lmy07b', etc., and unify them together. Such as
elif repo_type in ['irdakk', 'irdal', ...]:
    file_image = dir_root + '/out/dist/' + ...

File: python/aosp.py [hunk 3]
```diff
@@ -463,15 +472,29 @@ def flash_image():
         fileinput.close()
 
     # enter fastboot mode
-    android_enter_fastboot(device_id=device_id)
+    if repo_type == 'stable':
+        if not os.path.exists(dir_server_workaround):
+            error('Please add {flash.sh, fastboot.img, loader.efi} into %s!' % dir_server_workaround)
+
+        # enter dnx mode for ecs
+        android_enter_dnx(device_id=device_id)
```
Comment (gyagp) [inline python/aosp.py]: Remove the comment as it will not be limited to ecs.

File: python/aosp.py [hunk 4]
```diff
@@ -669,56 +698,35 @@ def _sync_repo(dir, cmd):
...
-            combo_prefix = 'ecs_e7'
-            combo_suffix = '-' + variant
-
-            if device_arch == 'x86_64':
-                combo = combo_prefix + '_64p' + combo_suffix
-            elif device_arch == 'x86':
-                combo = combo_prefix + combo_suffix
+        return 'aosp_' + device_type
+    elif repo_type == 'irdakk' or repo_type == 'irdal':
+        product = 'coho'
```
Comment (gyagp) [inline python/aosp.py]: We also need to use product_brand and product_name for irda.

File: python/aosp.py [hunk 5]
```diff
@@ -669,56 +698,35 @@ def _sync_repo(dir, cmd):
...
-                if device_arch == 'x86_64':
-                    combo = combo_prefix + device_type + '_64p' + combo_suffix
-                elif device_arch == 'x86':
-                    combo = combo_prefix + device_type + combo_suffix
+                product = 'aosp_' + device_type
 
-    return combo
+    if device_arch == 'x86_64':
+        if repo_type == 'irdal':
+            return product + 'l'
```
Comment (gyagp) [inline python/aosp.py]: Don't return early here as we may add more logic behind this snippet of code.

File: python/util.py [hunk 6]
```diff
@@ -1351,8 +1352,16 @@ def android_config_device(device_id, device_product, default, governor='', freq=
         info('Set governor to %s and freq to %s' % (governor, freq))
 
 
+def android_enter_dnx(device_id):
+    execute('timeout 6s ' + adb(cmd='reboot dnx', device_id=device_id))
+    # because dnx mode is basic, assume entering dnx mode will success anytime
```
Comment (gyagp) [inline python/util.py]: success -> succeed

File: python/aosp.py [hunk 7]
```diff
@@ -669,56 +698,35 @@ def _sync_repo(dir, cmd):
...
+        if device_type == 'baytrail':
             if ver_cmp(repo_ver, '2.0') >= 0:
-                combo_prefix = 'asus_t100'
-                combo_suffix = '-' + variant
-
-                if device_arch == 'x86_64':
-                    combo = combo_prefix + '_64p' + combo_suffix
-                elif device_arch == 'x86':
-                    combo = combo_prefix + combo_suffix
+                product = '%s_%s' % (product_brand, product_name)
```
Comment (gyagp) [inline python/aosp.py]: Remove this check

**Comment by gyagp** [comment]
Unify get_product in util.py and the one in aosp.py.

File: python/util.py [hunk 8]
```diff
@@ -179,6 +179,7 @@ def _get_real_dir(path):
 dir_workspace = '/workspace'
 dir_server = dir_workspace + '/server'
 dir_server_aosp = dir_server + '/aosp'
+dir_server_workaround = dir_server + '/workaround'
```
Comment (gyagp) [inline python/util.py]: Don't make this a global. Put it into share/python/aosp/workaround

File: python/util.py [hunk 9]
```diff
@@ -1055,7 +1061,8 @@ def analyze_issue(dir_aosp='/workspace/project/aosp-stable', dir_chromium='/work
     if device_id == '' or device_id == '192.168.42.1:5555':
         device_type = 'baytrail'
     target_arch = android_get_target_arch(device_id=device_id)
-    product = get_product(target_arch, device_type, ver)
+    # TODO(guanxian): will modify the call of get_product() since it is changed.
```
Comment (gyagp) [inline python/util.py]: Please make the change in this PR.

File: python/util.py [hunk 10]
```diff
@@ -976,19 +976,25 @@ def execute_adb_shell(cmd, device_id='', su=False, abort=False, show_cmd=False):
         return True
 
 
-def get_product(arch, device_type, ver):
+def get_product(repo_type, device_arch, device_type, product):
```
Comment (gyagp) [inline python/util.py]: This product is different with the returned one. Pick another better name. I also don't think putting product_brand and product_name in tuple is a good idea.

**Comment by gyagp** [comment]
Some more comments on commit message: 1) What's producti? 2) Please change the directory of workaround.

File: python/aosp.py [hunk 11]
```diff
@@ -572,7 +592,8 @@ def analyze():
     if len(devices_type) > 1 or devices_type[0] != 'baytrail':
         error('Only baytrail is supported to analyze')
 
-    analyze_issue(dir_aosp=dir_root, type=args.analyze_type, ver=repo_ver)
+    analyze_issue(dir_aosp=dir_root, type=args.analyze_type, device_id=devices_id[0],
+            repo_type=repo_type, device_type=devices_type[0], product_brand=product_brand, product_name=product_name)
```
Comment (legendlee1314) [inline python/aosp.py]: @gyagp , what do u think about this function with many parameters? in same line or split it like this now?

**Comment by legendlee1314** [comment]
@gyagp , i have updated commit info. please review again.

**Pull request merged**